azurelinuxagent/common/osutil/debian.py

```diff
@@ -33,10 +33,10 @@
 from azurelinuxagent.common.osutil.default import DefaultOSUtil
 
 
-class DebianOSUtil(DefaultOSUtil):
+class DebianOSBaseUtil(DefaultOSUtil):
 
     def __init__(self):
-        super(DebianOSUtil, self).__init__()
+        super(DebianOSBaseUtil, self).__init__()
```

Comment:
OK, I have to blow the whistle here. This use of super() is completely wrong-headed. I know you didn't introduce it, but it's just plain wrong.

If the class hierarchy is always single-inheritance, and I'm pretty sure it is, the correct way to invoke the superclass constructor is explicitly, by it's name.
```
class DebianOSBaseUtil(DefaultOSUtil):
    def __init__(self):
        DebianOSUtil.__init__()
```
If `DebianOSUtil` inherits from a superclass, then `DebianOSUtil.__init__()` should explicitly invoke the __init__ method in its own superclass.

I recognize I just asked for a change in every osutil class. However, this abuse/misuse of `super()` is pretty egregious. Had this stuff been done correctly the first time, renaming the classes would have required two mouseclicks in PyCharm.

Reply:
Let's create an issue and address this in a separate PR to keep this PR more focused on gen2 resource disk fixing.